example/map_pt_space.cpp

```diff
@@ -0,0 +1,288 @@
+            printf("# Root finder failed! "
+                   "alpha1 alpha2, rho1, rho2, sie1(rho1,Tguess), sie2(rho2,Tguess) "
+                   "sietot = "
+                   "%.14e %.14e %.14e %.14e %.14e %.14e %.14e\n",
+                   alpha1, alpha2, rho1, rho2,
+                   eos1.InternalEnergyFromDensityTemperature(rho1, Tguess),
+                   eos2.InternalEnergyFromDensityTemperature(rho2, Tguess), sietot);
+            Ts(i) = Ps(i, 0) = Ps(i, 1) = -1;
+            trajRes(i, 0) = trajRes(i, 1) = trajRes(i, 2) = -1;
+          } else {
+            // compute the pressures
+            Ps(i, 0) = eos1.PressureFromDensityTemperature(rho1, Ts(i));
+            Ps(i, 1) = eos2.PressureFromDensityTemperature(rho2, Ts(i));
+
+            trajRes(i, 0) = alpha1 + alpha2 - 1.0;
+
+            const Real u1 =
+                rhobar1 * eos1.InternalEnergyFromDensityTemperature(rho1, Ts(i));
+            const Real u2 =
+                rhobar2 * eos2.InternalEnergyFromDensityTemperature(rho2, Ts(i));
+            trajRes(i, 1) = singularity::robust::ratio(u1 + u2 - utot, utot);
+            trajRes(i, 2) = singularity::robust::ratio(Ps(i, 1) - Ps(i, 0), utot);
+          }
+        });
+    // wait for completion
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+    Kokkos::fence();
+#endif
+
+    // Copy the Ts array to host
+    std::unique_ptr<DataBox, DBDeleter> pTs_h(
+        new DataBox(Spiner::AllocationTarget::Host, nvfrac));
+    DataBox Ts_h = *pTs_h;
+    portableCopyToHost(Ts_h.data(), Ts.data(), Ts.sizeBytes());
+
+    // Get max temperature achieved. Note this could have been done in
+    // a single step via a portableReduce, but we split it out for clarity.
+    Real Tmax_walked = Ts_h.max();
+
+    // Let's make a T grid
+    const Real nT = nvfrac + 1;
+
+    RegularGrid1D gTs(Tmin, Tmax_walked, nT);
+
+    // Now we can do one more loop where we compute the energy and
+    // pressure residuals as a function of T and alpha
+    std::unique_ptr<DataBox, DBDeleter> p_residuals(
+        new DataBox(Spiner::AllocationTarget::Device, nT, nvfrac, NEOS));
+    DataBox residuals = *p_residuals;
+
+    portableFor(
+        "Compute residual", 0, nT, 0, nvfrac, PORTABLE_LAMBDA(const int j, const int i) {
+          // Temperature
+          const Real T = gTs.x(j);
+
+          // volume fractions
+          const Real lvfrac = lvfracs.x(i);
+          const Real vfrac = std::pow(10., lvfrac);
+          const Real alpha1 = vfrac;
+          const Real alpha2 = 1. - vfrac;
+
+          // microphysical densities
+          Real rho1 = singularity::robust::ratio(rhobar1, alpha1);
+          Real rho2 = singularity::robust::ratio(rhobar2, alpha2);
+
+          const Real u1 = rhobar1 * eos1.InternalEnergyFromDensityTemperature(rho1, T);
+          const Real u2 = rhobar2 * eos2.InternalEnergyFromDensityTemperature(rho2, T);
+          residuals(j, i, 0) = u1 + u2 - utot, utot;
```

Comment:
Is there some weird application of the `,` operator here?

Reply:
I'm shocked that compiled lol. That should just be `residuals(j, i, 0) = u1 + u2 - utot;`. Thanks for the catch!